src/search/json_display.ml

```diff
@@ -0,0 +1,124 @@
+module Html = Tyxml.Html
+
+let display_expression_rhs args res =
+  let open Odoc_model.Lang in
+  match res with
+  | Some res -> (
+      " : "
+      ^
+      match args with
+      | TypeDecl.Constructor.Tuple args ->
+          let type_ =
+            match args with
+            | _ :: _ :: _ -> TypeExpr.(Arrow (None, Tuple args, res))
+            | [ arg ] -> TypeExpr.(Arrow (None, arg, res))
+            | _ -> res
+          in
+          Render.text_of_type type_
+      | TypeDecl.Constructor.Record fields ->
+          let fields = Render.text_of_record fields in
+          let res = Render.text_of_type res in
+          fields ^ " -> " ^ res)
+  | None -> (
+      match args with
+      | TypeDecl.Constructor.Tuple args -> (
+          match args with
+          | _ :: _ :: _ -> " of " ^ Render.text_of_type (TypeExpr.Tuple args)
+          | [ arg ] -> " of " ^ Render.text_of_type arg
+          | _ -> "")
+      | TypeDecl.Constructor.Record fields ->
+          let fields = Render.text_of_record fields in
+          " of " ^ fields)
+let display_constructor_type args res =
+  let open Odoc_model.Lang in
+  match args with
+  | TypeDecl.Constructor.Tuple args ->
+      let type_ =
+        match args with
+        | _ :: _ :: _ -> TypeExpr.(Arrow (None, Tuple args, res))
+        | [ arg ] -> TypeExpr.(Arrow (None, arg, res))
+        | _ -> res
+      in
+      Render.text_of_type type_
+  | TypeDecl.Constructor.Record fields ->
+      let fields = Render.text_of_record fields in
+      let res = Render.text_of_type res in
+      fields ^ " -> " ^ res
+
+let of_entry ({ id; doc; extra } : Entry.t) : Odoc_html.Json.json =
+  let j_url = `String (Render.url id) in
+  let kind_s =
+    match extra with
+    | TypeDecl _ -> "type"
+    | Module -> "module"
+    | Value _ -> "val"
+    | Doc _ -> "doc"
+    | Exception _ -> "exn"
+    | Class_type _ -> "class type"
+    | Method _ -> "method"
+    | Class _ -> "class"
+    | TypeExtension _ -> "type ext"
+    | ExtensionConstructor _ -> "extension constructor"
+    | ModuleType -> "module type"
+    | Constructor _ -> "constructor"
+    | Field _ -> "field"
+  in
+  let kind_html =
+    Html.span ~a:[ Html.a_class [ "entry-kind" ] ] [ Html.txt kind_s ]
+  in
+  let rhs =
+    let s =
+      match extra with
+      | TypeDecl { canonical = _; equation = _; representation = _; txt } ->
+          let segments = String.split_on_char '=' txt in
+          if List.length segments > 1 then
+            segments |> List.tl |> String.concat "=" |> String.trim
+            |> ( ^ ) " = " |> Option.some
+          else None
+      | Constructor { args; res } ->
+          Some (" : " ^ display_constructor_type args res)
+      | Field { mutable_ = _; type_; parent_type = _ } ->
```

Comment:
If you take the habit of priming (`'`) identifiers that conflict with keywords rather than suffixing them `_` these kind of patterns look much less confusing: 
```ocaml
| Field { mutable' = _; type'; parent_type = _ } ->